src/cargo/sources/path.rs

```diff
@@ -610,11 +610,11 @@ fn list_files_gix(
 
     let pathspec = {
         // Include the package root.
-        let mut include = BString::from(":/");
+        let mut include = BString::from(":(top)");
         include.push_str(package_prefix.as_ref());
 
         // Exclude the target directory.
-        let mut exclude = BString::from(":!/");
+        let mut exclude = BString::from(":!(exclude,top)");
         exclude.push_str(target_prefix.as_ref());
```

Comment:
Could you add a test for this?

Ideally this PR would be split up such that
- The first commit adds the test, showing the bad behavior
- This commit updates the test, showing the good behavior

See also https://doc.crates.io/contrib/process/working-on-cargo.html#submitting-a-pull-request